packages/quantic/force-app/main/default/lwc/quanticUtils/__tests__/quanticUtils.test.js

```diff
@@ -160,4 +164,40 @@ describe('c/quanticUtils', () => {
       });
     });
   });
+
+  describe('buildTemplateTextFromResult', () => {
+    const testTemplate =
+      '<a href="${clickUri}">${title}</a><br/><br/><quote>${excerpt}</quote>';
+    const testResult = {
+      clickUri: 'https://www.example.com',
+      title: 'Example Title',
+      excerpt: 'This is an example excerpt.',
+    };
+
+    it('should return a string with the result values inserted into the template', () => {
+      // @ts-ignore
+      expect(buildTemplateTextFromResult(testTemplate, testResult)).toBe(
+        `<a href="${testResult.clickUri}">${testResult.title}</a><br/><br/><quote>${testResult.excerpt}</quote>`
+      );
+    });
+
+    it('should return an empty string if the template is undefined or falsy', () => {
+      // @ts-ignore
+      expect(buildTemplateTextFromResult(undefined, testResult)).toBe('');
+    });
+
+    it('should not return the template field name as placeholder if the result does not contain the field', () => {
+      const resultWithoutExcerpt = {
+        clickUri: 'https://www.example.com',
+        title: 'Example Title',
+      };
+
+      // @ts-ignore
```

Comment:
why do we need the `@ts-ignore`

Reply:
Because the `buildTemplateTextFromResult` is expecting a full `coveo.Result` type complete with all the required fields. I didn't want to spend the time to make a full Result mock for this unit test.

If we already have one somewhere I will gladly use it instead :)

Reply:
All that's really relevant for the test are the fields in the object I pass to the utils method. I'm wondering if the method shouldn't be changed to just expect an Object instead of a `coveo.Result`